pybamm/expression_tree/binary_operators.py

```diff
@@ -605,10 +622,11 @@ def __init__(self, left, right):
             left.domain != ["current collector"]
             and left.domain != ["negative particle"]
             and left.domain != ["positive particle"]
+            and left.domain != []
         ):
             raise pybamm.DomainError(
                 """left child domain must be 'current collector', 'negative particle',
-                or 'positive particle', not'{}""".format(
+                'positive particle', or '', not'{}""".format(
```

Comment:
I think (lower case) `outer` function should just return a multiply instead in the case where `left.domain = []`? or does this just not work?

Reply:
new primary broadcast sometimes forces an upper case `Outer` for symbols with domain `[]`